openapi/spec.json

```diff
@@ -1177,6 +1183,168 @@
+            "$ref": "#/components/responses/401"
+          },
+          "404": {
+            "$ref": "#/components/responses/404"
+          },
+          "429": {
+            "$ref": "#/components/responses/429"
+          },
+          "500": {
+            "$ref": "#/components/responses/500"
+          }
+        }
+      }
+    },
+    "/devices/{organizationId}/{deviceId}/renew-scap-keys": {
+      "post": {
+        "tags": [
+          "Devices"
+        ],
+        "summary": "Renew a device's SCAP keys",
+        "description": "Generate new random SCAP keys for one of your organization's devices - to replace a set of compromised keys, for instance.",
+        "responses": {
+          "200": {
+            "description": "Success",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object",
+                  "properties": {
+                    "downlink": {
+                      "type": "string"
+                    },
+                    "uplink": {
+                      "type": "string"
+                    }
+                  }
+                }
+              }
+            }
+          },
+          "401": {
+            "$ref": "#/components/responses/401"
+          },
+          "404": {
+            "$ref": "#/components/responses/404"
+          },
+          "429": {
+            "$ref": "#/components/responses/429"
+          },
+          "500": {
+            "$ref": "#/components/responses/500"
+          }
+        }
+      }
+    },
+    "/providers": {
+      "get": {
+        "tags": [
+          "Devices"
+        ],
+        "summary": "List providers",
+        "description": "List all the configured LoRaWAN and Sigfox providers.",
+        "responses": {
+          "200": {
+            "description": "Success",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/components/schemas/ProviderInArray"
+                  }
+                }
+              }
+            }
+          },
+          "401": {
+            "$ref": "#/components/responses/401"
+          },
+          "404": {
+            "$ref": "#/components/responses/404"
+          },
+          "429": {
+            "$ref": "#/components/responses/429"
+          },
+          "500": {
+            "$ref": "#/components/responses/500"
+          }
+        }
+      }
+    },
+    "/providers/{providerId}": {
+      "get": {
+        "tags": [
+          "Devices"
+        ],
+        "summary": "Get a provider",
+        "description": "Get the details for a provider given its ID.",
+        "responses": {
+          "200": {
+            "description": "Success",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/ProviderInArray"
```

Comment:
If there is no difference between the ressources returned in the list endpoint, I think the schema must renamed to `Provider`.

Reply:
ok